src/ports/postgres/modules/deep_learning/madlib_keras.py_in

```diff
@@ -520,21 +544,35 @@ def fit_transition(state, dependent_var, independent_var, dependent_var_shape,
         and only gets cleared in eval transition at the last row of the last iteration.
 
     """
-    if not independent_var or not dependent_var:
+    if not dependent_var_shape:
+        plpy.error("fit_transition called with no data")
+
+    if not prev_serialized_weights or not model_architecture:
         return state
```

Comment:
When # msts < # segs, any segments not currently training a model will be called with model_weights = model_arch = NULL.  Since those segments have nothing to do, we just return early and wait for the other segments to finish training.

Reply:
makes sense